python/pyspark/pandas/data_type_ops/base.py

```diff
@@ -65,6 +65,7 @@
 
 
 T_IndexOps = TypeVar("T_IndexOps", bound="IndexOpsMixin")
+IndexOpsLike = Union["Series", "Index"]
```

Comment:
Why do we still need `IndexOpsLike` since we have `T_IndexOps`?

Reply:
Good reference is here: https://stackoverflow.com/questions/58903906/whats-the-difference-between-a-constrained-typevar-and-a-union